src/pybind/rbd/rbd.pyx

```diff
@@ -5391,7 +5391,7 @@ cdef class LockOwnerIterator(object):
                 ret = rbd_lock_get_owners(image.image, &self.lock_mode,
                                           self.lock_owners,
                                           &self.num_lock_owners)
-            if ret >= 0:
+            if ret >= 0 or ret == -errno.ENOENT:
```

Comment:
Actually it crashes for me -- we can't just break `ret == -errno.ENOENT`, we also need to reset `self.num_lock_owners`, otherwise it would try to decode `self.lock_owners`.

Also, it would be very nice to add a simple test to test/pybind/test_rbd.py. It could call `list(image.lock_get_owners())` to test that is empty, then could acquire the lock (`image.lock_acquire(RBD_LOCK_MODE_EXCLUSIVE)`) and test `lock_get_owners` again.